### PR TITLE
RUN-2267: new local node executor feature flag

### PR DESCRIPTION
**Pull request by gschueler**
**Is this a bugfix, or an enhancement? Please describe.**
Updates the “newlocal” node executor, to work as a drop in replacement for the old “local” node executor

- Execution context (env vars) and all log output should match the previous implementation
- It has the benefit of being able to kill child processes when the local process finishes
- It introduces an experimental flag to disable env var inheritance from the server process

**Describe the solution you've implemented**

Make some behind the scenes changes to enable a feature flag that can make the “newlocal” the default local executor

`rundeck.feature.newLocalNodeExecutor.enabled=true` will set it as the default local executor

Add a new functional test step in the CircleCI workflow was added “Test Feature NewLocalExec”

This runs the same “ExecutionOutputSpec” but with the newLocalNodeExecutor feature flag enabled, to validate the output is the same

.. if there are some other functional tests that use local node executor in specific ways, we can enable those tests with this feature flag as well

**Describe alternatives you've considered**

**Additional context**

### CODE REVIEW

File: core/src/main/java/com/dtolabs/rundeck/core/utils/ScriptExecUtil.java [hunk 1]
```diff
@@ -248,14 +289,24 @@ public static int runLocalCommand(
...
+            }
             if(exec.isAlive()){
                 exec.waitFor();
             }
             throw new InterruptedException("Execution interrupted with code: " + exec.exitValue());
         }
     }
 
+    public static void killProcessHandleDescend(ProcessHandle handle) {
+        handle.descendants().forEach(ScriptExecUtil::killProcessHandleDescend);
```
Comment (ahormazabal) [inline core/src/main/java/com/dtolabs/rundeck/core/utils/ScriptExecUtil.java:303]: As discussed, just calling `ScriptExecUtil::killProcessHandle` should be enough here as `descendants()` already return children recursively.

**Review by ahormazabal — CHANGES_REQUESTED** [review]
Overall LGTM, only requesting minor changes.

Tested and works as expected, feature flag correctly switched the default executor.

Child Process killing tested OK, but won't work with processes that are run in background with &. 

As discussed, Runner is not supported yet but can be added in a next phase.

**Comment by ahormazabal** [comment]
> As discussed, Runner is not supported yet but can be added in a next phase.

Clarifying: Runned worked OK, but won't use the newlocal executor.

File: functional-test/src/test/resources/docker/compose/oss/docker-compose.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ services:
       RUNDECK_TOKENS_FILE: /home/rundeck/server/config/tokens.properties
       RUNDECK_MULTIURL_ENABLED: "true"
       RUNDECK_METRICS_ENABLED: "true"
+      RUNDECK_FEATURE_XYZ_ENABLED: "true"
```
[the remark this answers is not in the document]
Comment (gschueler) [reply functional-test/src/test/resources/docker/compose/oss/docker-compose.yml:14]: sure thing, sorry about that!

**Review by ahormazabal — APPROVED** [review]

**Pull request merged**